docs/source/clis.md

```diff
@@ -1,105 +1,224 @@
+
+<hfoptions id="launch_args">
+<hfoption id="SFT inline">
 
 ```bash
-trl sft --config examples/cli_configs/example_config.yaml --output_dir test-trl-cli --lr_scheduler_type cosine_with_restarts
+trl sft \
+  --model_name_or_path Qwen/Qwen2.5-0.5B \
+  --dataset_name stanfordnlp/imdb \
+  --num_processes 4
+```
+
+</hfoption>
+<hfoption id="SFT w/ config file">
+
+```yaml
+# sft_config.yaml
+model_name_or_path: Qwen/Qwen2.5-0.5B
+dataset_name: stanfordnlp/imdb
+num_processes: 4
 ```
 
-Will force-use `cosine_with_restarts` for `lr_scheduler_type`.
+Launch with:
 
-### Supported Arguments 
+```bash
+trl sft --config sft_config.yaml
+```
 
-We do support all arguments from `transformers.TrainingArguments`, for loading your model, we support all arguments from `~trl.ModelConfig`:
+</hfoption>
+<hfoption id="DPO inline">
 
-[[autodoc]] ModelConfig
+```bash
+trl dpo \
+  --model_name_or_path Qwen/Qwen2.5-0.5B \
+  --dataset_name anthropic/hh-rlhf \
+  --num_processes 4
+```
 
-You can pass any of these arguments either to the CLI or the YAML file.
+</hfoption>
+<hfoption id="DPO w/ config file">
 
-### Supervised Fine-tuning (SFT)
+```yaml
+# dpo_config.yaml
+model_name_or_path: Qwen/Qwen2.5-0.5B
+dataset_name: anthropic/hh-rlhf
+num_processes: 4
+```
 
-Follow the basic instructions above and run `trl sft --output_dir <output_dir> <*args>`: 
+Launch with:
 
 ```bash
-trl sft --model_name_or_path facebook/opt-125m --dataset_name stanfordnlp/imdb --output_dir opt-sft-imdb
+trl dpo --config dpo_config.yaml
 ```
+</hfoption>
+</hfoptions>
+
+### Using `--accelerate_config` for Accelerate Configuration
 
-The SFT CLI is based on the `trl/scripts/sft.py` script.
+The `--accelerate_config` flag lets you easily configure distributed training with [🤗 Accelerate](https://github.com/huggingface/accelerate). This flag accepts either:
 
-### Direct Policy Optimization (DPO)
+* the name of a predefined config profile (built into TRL), or
+* a path to a custom Accelerate YAML config file.
 
-To use the DPO CLI, you need to have a dataset in the TRL format such as 
+#### Predefined Config Profiles
 
-* TRL's Anthropic HH dataset: https://huggingface.co/datasets/trl-internal-testing/hh-rlhf-helpful-base-trl-style
-* TRL's OpenAI TL;DR summarization dataset: https://huggingface.co/datasets/trl-internal-testing/tldr-preference-trl-style
+TRL provides several ready-to-use Accelerate configs to simplify common training setups:
 
-These datasets always have at least three columns `prompt, chosen, rejected`:
+| Name              | Description                            |
+| ----------------- | -------------------------------------- |
+| `deepspeed_zero1` | DeepSpeed ZeRO Stage 1                 |
+| `deepspeed_zero2` | DeepSpeed ZeRO Stage 2                 |
+| `deepspeed_zero3` | DeepSpeed ZeRO Stage 3                 |
+| `fsdp_qlora`      | Fully Sharded Data Parallel with QLoRA |
```

Comment:
The QLoRA distinction is not really needed IMO and is an artifact from when we tuned L3 405B. I suggest we merge this https://github.com/huggingface/trl/pull/3317 first and then just have `fsdp1` and `fsdp2`

Reply:
Sounds good, I just rebased on `fix-fsdp2`, and making the required changes

Reply:
FSDP1/2 added in 2a52ee64261148260b8f08130634341e14e2a375